geomesa/src/main/scala/geotrellis/spark/io/geomesa/GeoMesaLayerWriter.scala

```diff
@@ -0,0 +1,41 @@
+package geotrellis.spark.io.geomesa
+
+import geotrellis.spark._
+import geotrellis.vector._
+import geotrellis.geotools._
+
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+import org.geotools.data.Transaction
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConversions._
+
+class GeoMesaLayerWriter(val attributeStore: GeoMesaAttributeStore, table: String)(implicit sc: SparkContext) extends Serializable {
+  def write[G <: Geometry, D: ? => Seq[(String, Any)]]
+    (layerId: LayerId, rdd: RDD[Feature[G, D]])
+    (implicit ev: Feature[G, D] => FeatureToGeoMesaSimpleFeatureMethods[G, D]): Unit = {
+
+    rdd
+      .zipWithIndex // it is necessary to provide unique id for ingest batch
+      .map { case (f, index) => val sf = f.toSimpleFeature(s"${attributeStore.layerIdString(layerId)}_${index}", layerId.name); sf.getFeatureType -> sf }.groupByKey
```

Comment:
@echeipesh the practise during one ingest batch features should have different ids; but they could be same during different batches; would ask this question in a geomesa channel to clarify it